### ISSUE TITLE
[Bug]: ValueError Gram matrix in 401k example

### ISSUE THREAD

**Issue by JohannWO** [DoubleML/doubleml-for-py#142]
### Describe the bug

https://docs.doubleml.org/stable/examples/py_double_ml_pension.html
example shows ValueError in PLR model
ValueError Gram matrix passed in via 'precompute' parameter did not pass validation when a single element was checked - please check that it was computed properly. For element (4,5) we computed 3375.771728515625 but the user-supplied value was 3375.773193359375.

### Minimum reproducible code snippet

https://docs.doubleml.org/stable/examples/py_double_ml_pension.html

# Initialize learners
Cs = 0.0001*np.logspace(0, 4, 10)
lasso = make_pipeline(StandardScaler(), LassoCV(cv=5, max_iter=10000))
lasso_class = make_pipeline(StandardScaler(),
                            LogisticRegressionCV(cv=5, penalty='l1', solver='liblinear',
                                                 Cs = Cs, max_iter=1000))
np.random.seed(123)
# Initialize DoubleMLPLR model
dml_plr_lasso = dml.DoubleMLPLR(data_dml_base,
                                ml_g = lasso,
                                ml_m = lasso_class,
                                n_folds = 3)
try:
    dml_plr_lasso.fit(store_predictions=True)
except ValueError as ve:
    print('ignore exception ValueError', ve)

### Expected Result

no ValueError

### Actual Result

ValueError Gram matrix passed in via 'precompute' parameter did not pass validation when a single element was checked - please check that it was computed properly. For element (4,5) we computed 3375.771728515625 but the user-supplied value was 3375.773193359375.

### Versions

Linux-5.13.0-30-generic-x86_64-with-glibc2.29
Python 3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0]
DoubleML 1.0 -> pip list says DoubleML 0.4.1
Scikit-Learn 1.0

**Comment by MalteKurz** [comment]
Thanks for your bug report. This is not a bug in `DoubleML` but in scikit-learn introduced in version `1.0.0` and not yet fixed. For further information see https://github.com/DoubleML/doubleml-docs/issues/45#issuecomment-930172645, https://github.com/scikit-learn/scikit-learn/issues/21997 and https://github.com/scikit-learn/scikit-learn/pull/22059.

As a temporary fix you can either downgrade to scikit-learn in version < 1.0.0 or you adapt the datatype in the pandas DataFrame in the following way (it then doesn't matter what scikit-learn version you use):

```python
import numpy as np

import doubleml as dml
from doubleml.datasets import fetch_401K
from sklearn.linear_model import LassoCV, LogisticRegressionCV
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import make_pipeline

data = fetch_401K(return_type='DataFrame')

# HERE: Change of datatypes
data = data.astype({'nifa': 'float64', 'net_tfa': 'float64', 'tw': 'float64', 'inc': 'float64'})

features_base = ['age', 'inc', 'educ', 'fsize', 'marr',
                 'twoearn', 'db', 'pira', 'hown']
data_dml_base = dml.DoubleMLData(data,
                                 y_col='net_tfa',
                                 d_cols='e401',
                                 x_cols=features_base)

Cs = 0.0001*np.logspace(0, 4, 10)
lasso = make_pipeline(StandardScaler(), LassoCV(cv=5, max_iter=10000))
lasso_class = make_pipeline(StandardScaler(),
LogisticRegressionCV(cv=5, penalty='l1', solver='liblinear',
Cs = Cs, max_iter=1000))

np.random.seed(123)
dml_plr_lasso = dml.DoubleMLPLR(data_dml_base,
ml_g = lasso,
ml_m = lasso_class,
n_folds = 3)
dml_plr_lasso.fit(store_predictions=True)
```


Btw, thanks for the "DoubleML 1.0 -> pip list says DoubleML 0.4.1" hint. This was in fact caused by a small bug in our issues template, see https://github.com/DoubleML/doubleml-for-py/commit/db6be27fa79837d2f5b3bf53f58f5d252d6d3895.

**Comment by JohannWO** [comment]
Hi Malte,
thanks for clarification. I was not able to detect this a scikit-learn error.
i applied your workaround and it works.
Thanks

**Comment by MalteKurz** [comment]
No Worries, that is often not so easy to identify. Thanks again for the report which hopefully helps other users facing the same issue.
